libraries/rush-lib/src/api/RushProjectSelector.ts

```diff
@@ -0,0 +1,183 @@
+    this._scopes.set(
+      'git',
+      new GitChangedProjectSelectorParser(this._rushConfig, this._options.gitSelectorParserOptions)
+    );
+    this._scopes.set('tag', new TagProjectSelectorParser(this._rushConfig));
+    this._scopes.set('version-policy', new VersionPolicyProjectSelectorParser(this._rushConfig));
+    this._scopes.set('json', new JsonFileSelectorParser(this._rushConfig, this));
+  }
+
+  /**
+   * Select a set of projects using the passed selector expression.
+   *
+   * The passed context string is used only when constructing error messages, in the event of
+   * an error in user input. The default string "expression" is used if no context is provided.
+   */
+  public async selectExpression(
+    expr: SelectorExpression,
+    context: string = 'expression'
+  ): Promise<RushConfigurationProject[]> {
+    if (isAnd(expr)) {
+      return this._evaluateAnd(expr, context);
+    } else if (isOr(expr)) {
+      return this._evaluateOr(expr, context);
+    } else if (isNot(expr)) {
+      return this._evaluateNot(expr, context);
+    } else if (isParameter(expr)) {
+      return this._evaluateParameter(expr, context);
+    } else if (isDetailedSelector(expr)) {
+      return this._evaluateDetailedSelector(expr, context);
+    } else if (typeof expr === 'string') {
+      return this._evaluateSimpleSelector(expr, context);
+    } else {
+      // Fail-safe... in general, this shouldn't be possible, as user script type checking
+      // or JSON schema validation should catch it before this point.
+      throw new SelectorError(`Invalid object encountered in selector expression in ${context}.`);
+    }
+  }
+
+  private async _evaluateAnd(
+    expr: IExpressionOperatorAnd,
+    context: string
+  ): Promise<RushConfigurationProject[]> {
+    const result: Array<RushConfigurationProject>[] = [];
+    for (const operand of expr.and) {
+      result.push(await this.selectExpression(operand, context));
+    }
+    return [...Selection.intersection(new Set(result[0]), ...result.slice(1).map((x) => new Set(x)))];
+  }
+
+  private async _evaluateOr(
+    expr: IExpressionOperatorOr,
+    context: string
+  ): Promise<RushConfigurationProject[]> {
+    const result: Array<RushConfigurationProject>[] = [];
+    for (const operand of expr.or) {
+      result.push(await this.selectExpression(operand, context));
+    }
+    return [...Selection.union(new Set(result[0]), ...result.slice(1).map((x) => new Set(x)))];
+  }
+
+  private async _evaluateNot(
+    expr: IExpressionOperatorNot,
+    context: string
+  ): Promise<RushConfigurationProject[]> {
+    const result: RushConfigurationProject[] = await this.selectExpression(expr.not, context);
+    return this._rushConfig.projects.filter((p) => !result.includes(p));
+  }
+
+  private async _evaluateParameter(
+    expr: ExpressionParameter,
+    context: string
+  ): Promise<RushConfigurationProject[]> {
+    const key: string = Object.keys(expr)[0];
+
+    if (key === '--to') {
```

Comment:
Shouldn't we still include `--impacted-by`, `--impacted-by-except`, and `--to-except`?